test/api_integration/apis/data_views/fields_for_wildcard_route/response.js

```diff
@@ -201,6 +201,14 @@ export default function ({ getService }) {
           indices: ['basic_index'],
         });
     });
+
+    it('returns 404 when neither exists', async () => {
+      await supertest
+        .get('/api/index_patterns/_fields_for_wildcard')
+        .query({ pattern: 'bad_index,bad_index_2' })
+        .expect(404);
```

Comment:
```suggestion
        .expect(200);
```
since it's no longer an error condition, this, with the reverting the change for `fields_for` should make this PR ready to merge